modules/targets/darwin/default.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.targets.darwin;
+
+  toDefaultsFile = domain: attrs:
+    pkgs.writeText "${domain}.plist" (lib.generators.toPlist { } attrs);
+
+  toActivationCmd = domain: attrs:
+    "$DRY_RUN_CMD defaults import ${escapeShellArg domain} ${
+      toDefaultsFile domain attrs
+    }";
+
+  nonNullDefaults =
+    mapAttrs (domain: attrs: (filterAttrs (n: v: v != null) attrs))
+    cfg.defaults;
+  writableDefaults = filterAttrs (domain: attrs: attrs != { }) nonNullDefaults;
+  activationCmds = mapAttrsToList toActivationCmd writableDefaults;
```

Comment:
I changed the logic here a little so that it won't perform a `defaults import` when the target domain includes only null values. So for example, when a user has the following configuration, it won't perform a `defaults import com.apple.desktopservices`:

```nix
{ "com.apple.desktopservices".DSDontWriteNetworkStores = null; }
```

The previous behavior should be harmless, but it might confuse users running Home Manager with verbose output.